python/cuml/internals/base.pyx

```diff
@@ -108,11 +117,16 @@ class Base(TagsMixin,
     verbose : int or boolean, default=False
         Sets logging level. It must be one of `cuml.common.logger.level_*`.
         See :ref:`verbosity-levels` for more info.
-    output_type : {'input', 'cudf', 'cupy', 'numpy', 'numba'}, default=None
-        Variable to control output type of the results and attributes of
-        the estimator. If None, it'll inherit the output type set at the
-        module level, `cuml.global_settings.output_type`.
+    output_type : {'input', 'array', 'dataframe', 'series', 'df_obj', 'numba', 'cupy', 'numpy', 'cudf', 'pandas'}, default=None
+        Return results and set estimator attributes to the indicated output
+        type. If None, the output type set at the module level
+        (`cuml.global_settings.output_type`) will be used.
         See :ref:`output-data-type-configuration` for more info.
+    output_mem_type : {'host', 'device'}, default=None
```

Comment:
I'm not sure that `output_mem_type` is a good name to describe what we mean here since it could be mistaken for type in the Python sense. Would `output_memory_location` or `output_memory_region` maybe be better?